test/ParserTest.php

```diff
@@ -5,15 +5,92 @@
 use Amp\Beanstalk\Parser;
 use PHPUnit\Framework\TestCase;
 
-class ParserTest extends TestCase {
-    public function testParsesPartialResponseCorrectly() {
-        $parser = new Parser(function ($result) use (&$parsed) {
-            $parsed = $result;
+class ParserTest extends TestCase
+{
+    /**
+     * @param mixed &$parsed_elements
+     * @return Parser
+     */
+    protected function getParserToTest(&$parsed_elements)
+    {
+        return new Parser(function ($result) use (&$parsed_elements) {
+            $parsed_elements = $result;
         });
+    }
+
+    public function testParsesPartialResponseCorrectly()
+    {
+        $parser = $this->getParserToTest($parsed);
```

Comment:
You can just create a property and define a `setUp()` method that creates a new parser.